modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesKernelBoot.java

```diff
@@ -264,20 +265,27 @@ private static AndesContextStore createAndesContextStoreFromConfig() throws Exce
      * @return an implementation of {@link MessageStore}
      * @throws Exception if an error occurs
      */
-    private static MessageStore createMessageStoreFromConfig(AndesContextStore andesContextStore) throws Exception {
-        StoreConfiguration andesConfiguration = AndesContext.getInstance()
-                .getStoreConfiguration();
+    private static MessageStore createMessageStoreFromConfig(AndesContextStore andesContextStore,
+                                                             FailureObservingStoreManager failureObservingStoreManager)
```

Comment:
FailureObservingStoreManager is used by the dtxStore which is created by the MessageStore. Therefore we need to pass it.